### PR TITLE
Checkout redesign

### PR DESCRIPTION
**Pull request by albertkol**
## Done

- Feature branch of all the Checkout Redesign work

## QA

- Are the cypress tests working? Those will cover trial and store purchases.
- Can you purchase renewals?
- Can you purchase offers?
- Can you purchase expired store purchases?

## Screenshot

Screenshot 2023-02-15 at 18-05-04 Buy Ubuntu Pro Ubuntu
Screenshot 2023-02-15 at 18-04-48 Buy Ubuntu Pro Ubuntu
Screenshot 2023-02-15 at 18-04-15 Buy Ubuntu Pro Ubuntu

### CODE REVIEW

File: static/js/src/advantage/subscribe/checkout/components/FreeTrial/FreeTrial.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+              )}{" "}
+              after which time you will be charged
+              {taxData?.total
+                ? ` ${currencyFormatter.format(Number(taxData?.total) / 100)}`
+                : ""}
+              .
+            </p>
+            <p>
+              You can cancel your subscription before your free trial ends in
+              your UA Dashboard
```
Comment (jpmartinspt) [inline static/js/src/advantage/subscribe/checkout/components/FreeTrial/FreeTrial.tsx:52]: `UA dashboard` -> `Ubuntu Pro dashboard`

File: static/js/src/advantage/subscribe/checkout/components/UserInfoForm/UserInfoForm.tsx [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+    }
+    return errorMessage;
+  };
+
+  const displayMode = (
+    <>
+      <Col size={12}>
+        <PaymentMethodSummary />
+      </Col>
+      {values.buyingFor === "organisation" ? (
```
Comment (jpmartinspt) [inline static/js/src/advantage/subscribe/checkout/components/UserInfoForm/UserInfoForm.tsx:166]: On my staging account I get my name as the organisation name, might be a bit unclear.

File: static/js/src/advantage/subscribe/checkout/components/UserInfoForm/UserInfoForm.tsx [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+      />
+    </>
+  );
+
+  return (
+    <Row>
+      {isEditing ? editMode : displayMode}
+      {window.accountId ? (
+        <div className="u-align--right">
+          <ActionButton onClick={toggleEditing} loading={isSubmitting}>
```
Comment (jpmartinspt) [inline static/js/src/advantage/subscribe/checkout/components/UserInfoForm/UserInfoForm.tsx:337]: Should we have a cancel button? When logged in and with payment info set if I click edit I loose the payment method details and there is no way to revert without refreshing.

**Comment by lyubomir-popov** [comment]
@albertkol looking mostly good, just a few small niggles listed here https://miro.com/app/board/uXjVONNsexk=/

File: static/js/src/advantage/subscribe/checkout/components/UserInfoForm/UserInfoForm.tsx (hunk 2 above)
Comment (albertkol) [reply static/js/src/advantage/subscribe/checkout/components/UserInfoForm/UserInfoForm.tsx:166]: I tested it and this is what I found.

1st the design is this:
image
Organisation name shows as an `h3`. Then your name I think is selected from the SSO data that gets passed in the macaroon.

This is what we have in the new checkout:
image

Quite similar to the design.

Not sure why you get your name as an organisation name. Could it be that the organisation name is your name?

File: static/js/src/advantage/subscribe/checkout/components/FreeTrial/FreeTrial.tsx (hunk 1 above)
Comment (albertkol) [reply static/js/src/advantage/subscribe/checkout/components/FreeTrial/FreeTrial.tsx:52]: Fixed.

File: static/js/src/advantage/subscribe/checkout/components/UserInfoForm/UserInfoForm.tsx (hunk 3 above)
Comment (albertkol) [reply static/js/src/advantage/subscribe/checkout/components/UserInfoForm/UserInfoForm.tsx:376]: Added cancel button.

**Review by jpmartinspt — APPROVED** [review]

**Pull request merged**